impl/src/main/java/org/ehcache/internal/copy/SerializingCopier.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.internal.copy;
+
+import org.ehcache.spi.copy.Copier;
+import org.ehcache.spi.serialization.Serializer;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * A helper copier implementation that performs the "copying" using serialization.
+ *
+ * @author Albin Suresh
+ */
+public final class SerializingCopier<T> implements Copier<T>, Serializer<T> {
+
+  private final Serializer<T> serializer;
+
+  public SerializingCopier(Serializer<T> serializer) {
+    this.serializer = serializer;
+  }
+
+  @Override
+  public T copyOnRead(final T obj) {
+    throw new AssertionError("This method should never be used.");
```

Comment:
Why? We will do our best to not use it, and I can see that behaviour in a test class. But not in a product class.
Same of `copyOnWrite`